samples/nrf9160/serial_lte_modem/src/slm_at_gps.c

```diff
@@ -52,6 +57,156 @@ void rsp_send(const u8_t *str, size_t len);
 /* global variable defined in different files */
 extern struct at_param_list at_param_list;
 extern char rsp_buf[CONFIG_AT_CMD_RESPONSE_MAX_LEN];
+extern struct k_work_q slm_work_q;
+
+#ifdef CONFIG_SUPL_CLIENT_LIB
+static int supl_fd;
+int connect_supl_server(void)
+{
+	int ret;
+	struct sockaddr_in server;
+	struct addrinfo *res;
+	struct addrinfo hints = {
+		.ai_family = AF_INET,
+		.ai_socktype = SOCK_STREAM,
+		.ai_protocol = IPPROTO_TCP,
+		/* Either a valid,
+		 * NULL-terminated access point name or NULL.
+		 */
+		.ai_canonname = NULL,
+	};
+
+	ret = socket(AF_INET, SOCK_STREAM, IPPROTO_TCP);
+	if (ret < 0) {
+		LOG_ERR("socket() error: %d", -errno);
+		return -errno;
+	}
+
+	supl_fd = ret;
+	ret = getaddrinfo(CONFIG_SLM_SUPL_SERVER, NULL, &hints, &res);
+	if (ret || res == NULL) {
+		LOG_ERR("getaddrinfo() error: %d", -errno);
```

Comment:
`getaddrinfo` returns an error through return value (`ret`). `errno` will only be set in case `EAI_SYSTEM` is returned.

Reply:
Thanks, fixed